src/FMU_Wrapper.f90

```diff
@@ -0,0 +1,753 @@
+!> @brief Initializes the FMU Slave
+!>
+!> @param self the FMU2_Slave to act on
+!> @param id Unique id for this FMU
+!> @param pList A parameter list with input options
+!>
+SUBROUTINE init_FMU2_Slave(self,id,pList)
+  CHARACTER(LEN=*),PARAMETER :: myName='init_FM2_Slave'
+  CLASS(FMU2_Slave),INTENT(INOUT) :: self
+  INTEGER(SIK),INTENT(IN) :: id
+  TYPE(ParamType),INTENT(IN) :: pList
+
+  ! Required FMU pList
+  IF(.NOT. pList%has('unzipDirectory')) CALL eFMU_Wrapper%raiseError(modName//'::'//myName//' - No FMU unzipDirectory')
+  CALL pList%get('unzipDirectory', self%unzipDirectory)
+  self%idFMU=id
+  self%FMU_version=2_SIK
+
+  ! Parse the FMU XML model description
+  CALL eFMU_Wrapper%raiseDebug('Opening FMU XML File: '//self%unzipDirectory//'/modelDescription.xml' )
+  CALL self%modelDescription%initFromXML(self%unzipDirectory//'/modelDescription.xml',.TRUE.)
+
+  ! Optional FMU pList
+  IF(pList%has('instanceName')) THEN
+    CALL pList%get('instanceName', self%instanceName)
+  ELSE
+    self%instanceName = "default_fmu_instance"
+  ENDIF
+  IF(pList%has('modelIdentifier')) THEN
+    CALL pList%get('modelIdentifier', self%modelIdentifier)
+  ELSE
+    CALL self%modelDescription%get('modelIdentifier', self%modelIdentifier)
+  ENDIF
+  IF(pList%has('guid')) THEN
+    CALL pList%get('guid', self%guid)
+  ELSE
+    CALL self%modelDescription%get('guid', self%guid)
+  ENDIF
+
+  ! Initilize the FMU
+  self%fmu_c_ptr = InitilizeFMU2_Slave(self%idFMU, &
+    CHAR(self%guid)//c_null_char, &
+    CHAR(self%modelIdentifier)//c_null_char, &
+    CHAR(self%unzipDirectory)//c_null_char, &
+    CHAR(self%instanceName)//c_null_char)
+
+  self%isInit=.TRUE.
+
+ENDSUBROUTINE
+!
+!-------------------------------------------------------------------------------
+!> @brief Set the FMU internal ODE integrator settings
+!>
+!> @param self the FMU2_Slave to act on
+!> @param toleranceDefined flag to override internal FMU ODE tolerance settings
+!> @param tolerance tolerance of FMU ODE integrator
+!> @param startTime start time of FMU ODE integration
+!> @param stopTimeDefined flag to determine if ODE integration stops at stopTime
+!> @param stopTime  end time of ODE integration
+!>
+SUBROUTINE setupExperiment_FMU2_Slave(self, toleranceDefined, tolerance, startTime, stopTimeDefined, stopTime)
+  CLASS(FMU2_Slave),INTENT(INOUT) :: self
+  LOGICAL(SBK),INTENT(IN) :: toleranceDefined
+  REAL(SRK),INTENT(IN) :: tolerance
+  REAL(SRK),INTENT(IN) :: startTime
+  LOGICAL(SBK),INTENT(IN) :: stopTimeDefined
+  REAL(SRK),INTENT(IN) :: stopTime
+
+  REQUIRE(self%isInit)
+  REQUIRE(c_associated(self%fmu_c_ptr))
+
+  CALL setupExperimentFMU2_Slave(self%fmu_c_ptr, LOGICAL(toleranceDefined,1), tolerance, startTime, &
+    LOGICAL(stopTimeDefined,1), stopTime)
+ENDSUBROUTINE
+!
+!-------------------------------------------------------------------------------
+!> @brief Get the valueReference that corresponds to the FMU variableName
```

Comment:
It is a handle to a FMU variable.  The interfaces defined by the [FMI standard](https://fmi-standard.org/docs/3.0-dev/) use the valueReference to get/set a variable in a FMU.  Since this is not very easy to use, the fortran FMU_Wrapper contains convenience methods to get/set variables by their variable name.

Reply:
updated doc string for clarity.